Content.Server/Polymorph/Systems/PolymorphableSystem.cs

```diff
@@ -0,0 +1,178 @@
+using Content.Server.Actions;
+using Content.Server.Buckle.Components;
+using Content.Server.Inventory;
+using Content.Server.Mind.Commands;
+using Content.Server.Mind.Components;
+using Content.Server.Polymorph.Components;
+using Content.Server.Popups;
+using Content.Shared.Actions;
+using Content.Shared.Actions.ActionTypes;
+using Content.Shared.Damage;
+using Content.Shared.Hands.EntitySystems;
+using Content.Shared.Polymorph;
+using Robust.Shared.Containers;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Polymorph.Systems
+{
+    public sealed partial class PolymorphableSystem : EntitySystem
+    {
+        [Dependency] private readonly ActionsSystem _actions = default!;
+        [Dependency] private readonly IEntityManager _entity = default!;
+        [Dependency] private readonly IPrototypeManager _proto = default!;
+        [Dependency] private readonly ServerInventorySystem _inventory = default!;
+        [Dependency] private readonly SharedHandsSystem _sharedHands = default!;
+        [Dependency] private readonly DamageableSystem _damageable = default!;
+        [Dependency] private readonly PopupSystem _popup = default!;
+        [Dependency] private readonly IMapManager _mapManager = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<PolymorphableComponent, ComponentStartup>(OnStartup);
+            SubscribeLocalEvent<PolymorphableComponent, PolymorphActionEvent>(OnPolymorphActionEvent);
+
+            InitializeMap();
+        }
+
+        private void OnStartup(EntityUid uid, PolymorphableComponent component, ComponentStartup args)
+        {
+            if (component.InnatePolymorphs != null)
+            {
+                foreach (var morph in component.InnatePolymorphs)
+                {
+                    CreatePolymorphAction(morph, uid);
+                }
+            }
+        }
+
+        private void OnPolymorphActionEvent(EntityUid uid, PolymorphableComponent component, PolymorphActionEvent args)
+        {
+            PolymorphEntity(uid, args.Prototype);
+        }
+
+        /// <summary>
+        /// Polymorphs the target entity into the specific polymorph prototype
+        /// </summary>
+        /// <param name="target">The entity that will be transformed</param>
+        /// <param name="id">The id of the polymorph prototype</param>
+        public void PolymorphEntity(EntityUid target, String id)
+        {
+            if (!_proto.TryIndex<PolymorphPrototype>(id, out var prototype))
+                return;
```

Comment:
Should just Index given linter should catch it. At a minimum if TryIndex was left it would need an error log.